### PR TITLE
add Prim instances for Int128 and Word128

### LINKED ISSUE #9: add `Prim` instances?
the `Prim` typeclass from the `primitive` package is useful for dealing with arrays of unboxed elements. Since the types in this library are just product types of unboxable things of the same type, having a `Prim` instance is safe, trivial to implement, and useful.

### PR DESCRIPTION
**Pull request by chessai**
resolves #9

### CODE REVIEW

File: src/Data/WideWord/Word128.hs [hunk 1]
```diff
@@ -434,6 +456,67 @@ pokeElemOff128 ptr idx (Word128 a1 a0) = do
...
+-- -----------------------------------------------------------------------------
+-- Functions for `Prim` instance.
+
+{-# INLINE sizeOf128# #-}
+sizeOf128# :: Word128 -> Int#
+sizeOf128# _ = 2# *# sizeOf# (undefined :: Word64)
+
+{-# INLINE alignment128# #-}
+alignment128# :: Word128 -> Int#
+alignment128# _ = alignment# (undefined :: Word64)
```
Comment (erikd) [inline src/Data/WideWord/Word128.hs]: Don't think that is correct :).

File: src/Data/WideWord/Word128.hs [hunk 2]
```diff
@@ -434,6 +456,67 @@ pokeElemOff128 ptr idx (Word128 a1 a0) = do
...
+
+{-# INLINE indexByteArray128# #-}
+indexByteArray128# :: ByteArray# -> Int# -> Word128
+indexByteArray128# arr# i# =
+  let x = indexByteArray# arr# (2# *# i#)
+      y = indexByteArray# arr# (2# *# i# +# 1#)
+  in Word128 x y
```
Comment (erikd) [inline src/Data/WideWord/Word128.hs]: Reading a array read from disk, that will give different results on a big endian vs a little endian machine.

Have a look at the way I did `peek128` and `poke128`.

File: src/Data/WideWord/Word128.hs (hunk 1 above)
Comment (chessai) [reply src/Data/WideWord/Word128.hs]: why not?

Comment (erikd) [reply src/Data/WideWord/Word128.hs]: Alignment for `Word128` should be `2 * alighnment (undefined :: Word64)` shouldn't it?

Comment (chessai) [reply src/Data/WideWord/Word128.hs]: Oh yeah. Oops

**Comment by erikd** [comment]
Thanks!

**Pull request merged**